core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup.go

```diff
@@ -217,21 +217,43 @@ func (r *EvmRegistry) allowedToUseMercury(opts *bind.CallOpts, upkeepId *big.Int
 		return encoding.NoPipelineError, encoding.UpkeepFailureReasonNone, false, allowed.(bool), nil
 	}
 
-	cfg, err := r.registry.GetUpkeepPrivilegeConfig(opts, upkeepId)
+	payload, err := r.packer.PackGetUpkeepPrivilegeConfig(upkeepId)
```

Comment:
let's ensure we get Felix's review on this file if we go ahead with this approach

Reply:
I'll ping him on it